### PR TITLE
web3 utils documentation

### LINKED ISSUE #4965: Web3 Utils Documentation

### PR DESCRIPTION
**Pull request by nikoulai**
## Description
closes #4965 

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-utils/src/converters.ts [hunk 1]
```diff
@@ -103,16 +113,42 @@ const bufferToHexString = (data: Buffer) => `0x${data.toString('hex')}`;
...
+ * ```ts
+ * console.log(web3.utils.hexToBytes('0x74657374'));
+ * > <Buffer 74 65 73 74>
+ * ```
  */
 export const hexToBytes = (bytes: HexString): Buffer => bytesToBuffer(bytes);
 
 /**
  * Converts value to it's number representation
+ * @param value - Value to be converted
```
Comment (luu-alex) [inline packages/web3-utils/src/converters.ts]:
```suggestion
 * @param value - Hex string to be converted
```

File: packages/web3-utils/src/converters.ts [hunk 2]
```diff
@@ -139,16 +183,33 @@ export const numberToHex = (value: Numbers): HexString => {
 };
 /**
  * Converts value to it's hex representation @alias `numberToHex`
+ *
  */
 export const fromDecimal = numberToHex;
 
 /**
  * Converts value to it's decimal representation in string
+ * @param value - Value to be converted
```
Comment (luu-alex) [inline packages/web3-utils/src/converters.ts]:
```suggestion
 * @param value - Hex string be converted
```

File: packages/web3-utils/src/hash.ts [hunk 3]
```diff
@@ -49,6 +49,17 @@ const SHA3_EMPTY_BYTES = '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfa
 
 /**
  * computes the Keccak-256 hash of the input and returns a hexstring
+ * @param data - the input to hash
+ * @returns - the Keccak-256 hash of the input
+ *
+ * @example
+ * ```ts
+ * console.loog(web3.utils.sha3('web3.js'));
```
Comment (luu-alex) [inline packages/web3-utils/src/hash.ts]:
```suggestion
 * console.log(web3.utils.sha3('web3.js'));
```

File: packages/web3-utils/src/hash.ts [hunk 4]
```diff
@@ -49,6 +49,17 @@ const SHA3_EMPTY_BYTES = '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfa
...
+ * > 0x63667efb1961039c9bb0d6ea7a5abdd223a3aca7daa5044ad894226e1f83919a
+ *
+ * console.loog(web3.utils.sha3(''));
```
Comment (luu-alex) [inline packages/web3-utils/src/hash.ts]:
```suggestion
 * console.log(web3.utils.sha3(''));
```

File: packages/web3-utils/src/hash.ts [hunk 5]
```diff
@@ -272,20 +318,26 @@ export const encodePacked = (...values: Sha3Input[]): string => {
...
+ * @returns - the keccack246 of the tightly packed values
  */
 export const soliditySha3 = (...values: Sha3Input[]): string | undefined =>
 	sha3(encodePacked(...values));
 
 /**
  * Will tightly pack values given in the same way solidity would then hash.
  * returns a hash string, if input is empty will return `0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`
+ * @param values - the input to return the tightly packed values
+ * @returns - the keccack246 of the tightly packed values
```
Comment (luu-alex) [inline packages/web3-utils/src/hash.ts:338]: Should have an example for this one

File: packages/web3-utils/src/hash.ts [hunk 6]
```diff
@@ -272,20 +318,26 @@ export const encodePacked = (...values: Sha3Input[]): string => {
 /**
  * Will tightly pack values given in the same way solidity would then hash.
  * returns a hash string, or null if input is empty
+ * @param values - the input to return the tightly packed values
+ * @returns - the keccack246 of the tightly packed values
```
Comment (luu-alex) [inline packages/web3-utils/src/hash.ts:322]: Should have an example for this one

**Comment by melissamforbs** [comment]
@dependabot merge

**Review by luu-alex — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

**Pull request merged**